src/features/todosState.ts

```diff
@@ -0,0 +1,55 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+import { mockedType } from '../screens/Home'
+export interface TodosState {
+  todos: mockedType[]
+}
+
+const initialState: TodosState = {
+  todos: [
+    { checked: false, description: 'hasdf', id: 0, title: 'pruebaRed' },
+    { checked: true, description: 'hasdf', id: 1, title: 'prueba2Red' },
+  ],
+}
+
+export const todosSlice = createSlice({
+  initialState,
+  name: 'todos',
+  reducers: {
+    addNewTodo: (state, action) => {
+      const { title, description } = action.payload
+      const idNewTodo = state.todos.length
+        ? state.todos[state.todos.length - 1].id + 1
+        : 0
+
+      const newTodo: mockedType = {
+        checked: false,
+        description,
+        id: idNewTodo,
+        title,
+      }
+      state.todos.push(newTodo)
+    },
+    check: (state, action) => {
+      const todoToCheck = state.todos.find(todo => todo.id === action.payload)
+      if (todoToCheck) {
+        todoToCheck.checked = true
+      }
+    },
```

Comment:
Nice one! Another workaround would be to name this `toggleCheck` and check the previous checked value. I'm not saying you need to change it, I like what you already did and it is a different approach, but I'm leaving here an alternative implementation just in case you want to have it in your toolchain.

```suggestion
    toggleCheck: (state, action) => {
      const todoToCheck = state.todos.find(todo => todo.id === action.payload)
      if (todoToCheck) {
        todoToCheck.checked = !todoToCheck.checked
      }
    },
```